src/Umbraco.Web.UI.Client/src/packages/core/repository/repository-details.manager.ts

```diff
@@ -0,0 +1,239 @@
+	entries = this.#entries.asObservable();
+	entryByUnique(unique: string) {
+		return this.#entries.asObservablePart((items) => items.find((item) => item.unique === unique));
+	}
+
+	#statuses = new UmbArrayState<UmbRepositoryStatus>([], (x) => x.unique);
+	statuses = this.#statuses.asObservable();
+
+	/**
+	 * Creates an instance of UmbRepositoryItemsManager.
+	 * @param {UmbControllerHost} host - The host for the controller.
+	 * @param {string} repository - The alias of the repository to use.
+	 * @memberof UmbRepositoryItemsManager
+	 */
+	constructor(host: UmbControllerHost, repository: UmbDetailRepository<DetailType> | string) {
+		super(host);
+
+		this.#entries.sortBy((a, b) => {
+			const uniques = this.getUniques();
+			const aIndex = uniques.indexOf(a.unique);
+			const bIndex = uniques.indexOf(b.unique);
+			return aIndex - bIndex;
+		});
+
+		if (typeof repository === 'string') {
+			this.#init = new UmbExtensionApiInitializer<ManifestRepository<UmbDetailRepository<DetailType>>>(
+				this,
+				umbExtensionsRegistry,
+				repository,
+				[this],
+				(permitted, repository) => {
+					this.repository = permitted ? repository.api : undefined;
+				},
+			).asPromise();
+		} else {
+			this.repository = repository;
+			this.#init = Promise.resolve();
+		}
+
+		this.observe(
+			this.uniques,
+			(uniques) => {
+				// remove entries based on no-longer existing uniques:
+				const removedEntries = this.#entries.getValue().filter((entry) => !uniques.includes(entry.unique));
+				this.#entries.remove(removedEntries);
+				this.#statuses.remove(removedEntries);
+				removedEntries.forEach((entry) => {
+					this.removeUmbControllerByAlias('observeEntry_' + entry.unique);
+				});
+
+				this.#requestNewItems();
+			},
+			null,
+		);
+
+		this.consumeContext(UMB_ACTION_EVENT_CONTEXT, (context) => {
+			this.#eventContext?.removeEventListener(
+				UmbEntityUpdatedEvent.TYPE,
+				this.#onEntityUpdatedEvent as unknown as EventListener,
+			);
+
+			this.#eventContext = context;
+			this.#eventContext.addEventListener(
+				UmbEntityUpdatedEvent.TYPE,
+				this.#onEntityUpdatedEvent as unknown as EventListener,
+			);
+		});
+	}
+
+	clear(): void {
+		this.#uniques.setValue([]);
+		this.#entries.setValue([]);
+		this.#statuses.setValue([]);
+	}
+
+	getUniques(): Array<string> {
+		return this.#uniques.getValue();
+	}
+
+	setUniques(uniques: string[] | undefined): void {
+		this.#uniques.setValue(uniques ?? []);
+	}
+
+	addUnique(unique: string): void {
+		this.#uniques.appendOne(unique);
+	}
+
+	addEntry(data: DetailType): void {
+		const unique = data.unique;
+		this.#statuses.appendOne({
+			state: {
+				type: 'success',
+			},
+			unique,
+		});
+		this.#entries.appendOne(data);
+		this.#uniques.appendOne(unique);
+		// Notice in this case we do not have a observable from the repo, but it should maybe be fine that we just listen for ACTION EVENTS.
+	}
+
+	getItems(): Array<DetailType> {
```

Comment:
Is it on purpose that you use the word "items" here? I can also see it used in some private methods. It might just be a copy/paste leftover, but I just wanted to make sure since this is a public method.

Reply:
No, this is some of the old varry over code from previous PR. Most likely by Lee. So feel free to adjust :)

Reply:
I have updated it to `getEntries`, as there already is an `addEntry` and `entryByUnique`. We can look into a base class at some point to align the two managers.